src/main/java/org/agrona/concurrent/ringbuffer/OneToOneRingBuffer.java

```diff
@@ -132,13 +133,15 @@ public boolean write(final int msgTypeId, final DirectBuffer srcBuffer, final in
 
         if (0 != padding)
         {
+            buffer.putLong(0, 0L);
             buffer.putLongOrdered(recordIndex, makeHeader(padding, PADDING_MSG_TYPE_ID));
             recordIndex = 0;
         }
 
         buffer.putBytes(encodedMsgOffset(recordIndex), srcBuffer, srcIndex, length);
+        buffer.putLong(recordIndex + alignedRecordLength, 0L);
```

Comment:
Thanks @nitsanw.
I'm okay with either approach, let me know if you need a change to the code.